src/AtomShell/process.jl

```diff
@@ -117,14 +117,32 @@ end
 
 export active
 
-active(shell::Electron) = process_running(shell.proc)
+Base.close(shell::Electron) = close(shell.sock)
+Base.isopen(shell::Electron) = process_running(shell.proc)
 
-quit(shell::Electron) = close(shell.sock)
+# TODO: Deprecate these functions in favor of methods above.
+active(shell::Electron) = isopen(shell)
+quit(shell::Electron) = close(shell)
 
 # Default process
 
-function shell(; debug = false)
-  global _shell
-  _shell ≠ nothing && active(_shell) && return _shell
-  _shell = init(debug = debug)
+_active_electron_process = Ref{Union{Electron, Nothing}}(nothing)
```

Comment:
This can be marked `const` for a significant performance improvement. I also personally prefer `ALL_CAPS` for global constants, but if you're happy with `_leading_underscore` notation that's fine.

Reply:
Yep, thanks.

I think I personally like the lowercase here because it's `const` but not immutable (though I think I've been inconsistent with that in other places).

Reply:
:) Don't forget to make this const before merging
```suggestion
const _active_electron_process = Ref{Union{Electron, Nothing}}(nothing)
```